### PR TITLE
feat: extract transform error logic to function

### PR DESCRIPTION
**Pull request by edikshsh**
feat: reduce number of new transforms

### CODE REVIEW

File: src/streams/transforms/base/simple-async-transform.ts [hunk 1]
```diff
@@ -18,7 +19,7 @@ export class SimpleAsyncTransform<TSource, TDestination> extends BaseTransform<T
     async _transform(chunk: TSource, encoding: BufferEncoding, callback: TypedTransformCallback<TDestination>) {
         const chunkClone = cloneDeep(chunk);
         try {
-            const result = await this.transformer(chunk);
+            const result = await this.transformer(chunk, this as Transform);
```
Comment (WilliamAlber) [inline src/streams/transforms/base/simple-async-transform.ts]: Why is the casting necessary here?

File: src/streams/transforms/utility/array-join-transform.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import { TransformCallback, TransformOptions } from 'stream';
 import { BaseTransform } from '../base/base-transform';
+import { SimpleTransform, TransformFunction } from '../base/simple-transform';
+import { FullTransformOptions } from '../types/full-transform-options.type';
 import { TypedTransformCallback } from '../types/typed-transform-callback';
 
 export class ArrayJoinTransform<TSource> extends BaseTransform<TSource, TSource[]> {
```
Comment (WilliamAlber) [inline src/streams/transforms/utility/array-join-transform.ts]: Delete due to refactor as the subsequent function fulfills this logic.

File: test-packages/easier-streams/test3.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import fs from 'fs'
```
Comment (WilliamAlber) [inline test-packages/easier-streams/test3.ts]: Delete this file.

**Review by WilliamAlber — CHANGES_REQUESTED** [review]

File: src/streams/transforms/base/simple-async-transform.ts (hunk 1 above)
Comment (edwardcym) [reply src/streams/transforms/base/simple-async-transform.ts]: Very strange reason.
`Transform`'s `_transform` function is not async, but here it is declare as async. This made typescript think for some reason that the `this` is not a `Transform`

I solved it by replacing the `async` syntax with `promise` syntax.
All tests passed so looks good.

File: src/streams/transforms/utility/array-join-transform.ts (hunk 2 above)
Comment (edwardcym) [reply src/streams/transforms/utility/array-join-transform.ts]: done

File: test-packages/easier-streams/test3.ts (hunk 3 above)
Comment (edwardcym) [reply test-packages/easier-streams/test3.ts]: done

**Review by WilliamAlber — APPROVED** [review]

**Pull request merged**